src/screens/SettingsSheet/components/GoogleAccountSection.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react';
+import {
+  getGoogleAccountUserData,
+  GoogleDriveUserData,
+  logoutFromGoogleDrive,
+} from '@/handlers/cloudBackup';
+import ImageAvatar from '@/components/contacts/ImageAvatar';
+import { showActionSheetWithOptions } from '@/utils';
+import * as i18n from '@/languages';
+import {
+  clearAllWalletsBackupStatus,
+  updateWalletBackupStatusesBasedOnCloudUserData,
+} from '@/redux/wallets';
+import { useDispatch } from 'react-redux';
+import Menu from './Menu';
+import MenuItem from './MenuItem';
+import { logger, RainbowError } from '@/logger';
+
+export const GoogleAccountSection: React.FC = () => {
+  const dispatch = useDispatch();
+  const [accountDetails, setAccountDetails] = useState<
+    GoogleDriveUserData | undefined
+  >(undefined);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getGoogleAccountUserData()
+      .then(accountDetails => {
+        setAccountDetails(accountDetails ?? undefined);
+      })
+      .catch(error => {
+        logger.error(
+          new RainbowError(
+            `Something went wrong when trying to fetch google account data to display in Backups Section, error: ${error?.toString()}`
+          )
+        );
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const removeBackupStateFromAllWallets = async () => {
+    setLoading(true);
+    await dispatch(clearAllWalletsBackupStatus());
+    setLoading(false);
+  };
+
+  const onGoogleAccountPress = () => {
+    showActionSheetWithOptions(
+      {
+        cancelButtonIndex: 2,
+        options: [
+          i18n.t(i18n.l.settings.backup_switch_google_account),
+          i18n.t(i18n.l.settings.backup_sign_out),
+          i18n.t(i18n.l.button.cancel),
+        ],
+      },
+      (buttonIndex: number) => {
+        if (buttonIndex === 0) {
+          logoutFromGoogleDrive();
+          setAccountDetails(undefined);
+          removeBackupStateFromAllWallets().then(() => loginToGoogleDrive());
+        } else if (buttonIndex === 1) {
+          logoutFromGoogleDrive();
+          setAccountDetails(undefined);
+          removeBackupStateFromAllWallets();
+        }
+      }
+    );
+  };
+
+  const loginToGoogleDrive = async () => {
+    setLoading(true);
+    await dispatch(updateWalletBackupStatusesBasedOnCloudUserData());
+    try {
+      const accountDetails = await getGoogleAccountUserData();
+      setAccountDetails(accountDetails ?? undefined);
+    } catch (error) {
+      logger.error(
+        new RainbowError(
+          `Something went wrong when logging into Google Drive, error: ${error?.toString()}`
+        )
```

Comment:
Nit: same thing. Another smol tip: the `Something went wrong` part isn't super helpful. What we really want is context on where the error was handled and any other relevant data that might help us debug.